include/openssl/core_names.h

```diff
@@ -75,18 +75,14 @@ extern "C" {
 #define OSSL_MAC_PARAM_SALT         "salt"      /* octet string */
 #define OSSL_MAC_PARAM_XOF          "xof"       /* int, 0 or 1 */
 #define OSSL_MAC_PARAM_FLAGS        "flags"     /* int */
-/* Note that "md" and "digest" are equivalent */
-#define OSSL_MAC_PARAM_MD           "md"        /* utf8 string */
-#define OSSL_MAC_PARAM_DIGEST       "digest"    /* utf8 string */
-#define OSSL_MAC_PARAM_CIPHER       "cipher"    /* utf8 string */
-/* Note that "algorithm" can be used instead of "md", "digest" or "cipher" */
+/*
+ * If "engine" or "properties" are specified, they should always be paired
+ * with "algorithm".
+ */
 #define OSSL_MAC_PARAM_ALGORITHM    "algorithm" /* utf8 string */
 #define OSSL_MAC_PARAM_ENGINE       "engine"    /* utf8 string */
 #define OSSL_MAC_PARAM_PROPERTIES   "properties" /* utf8 string */
-/* Note that "size", "digestsize" and "outlen" are equivalent */
 #define OSSL_MAC_PARAM_SIZE         "size"      /* size_t */
```

Comment:
None of these is an input size?

Reply:
Not as far as I know...  The only that actually existed in the old EVP_PKEY method form was "digestsize".  "size" was named such to correspond to the `EVP_MAC_size()` function, and "outlen" was my invention because I thought it was more expressive than "size", but I've realised that it would be an uphill battle to have that adopted, so I have it up.